### PR TITLE
Fixes normalize SES notification email addresses before saving

### PR DESCRIPTION
**Pull request by albertisfu**
Yesterday I detected an issue related to the new retry failed email function.

When designing the function, I supposed that every email address for a bounce or complaint notification always will have the format:

```
"bouncedRecipients":[
{
"status":"5.0.0",
"action":"failed",
"diagnosticCode":"smtp; 550 user unknown",
"emailAddress":"recipient1@example.com"
}
```

Where the email address is normalized, without a user name like `User Name <recipient1@example.com>`

As described in documentation: [https://docs.aws.amazon.com/ses/latest/dg/notification-contents.html#bounced-recipients](https://docs.aws.amazon.com/ses/latest/dg/notification-contents.html#bounced-recipients)

However seems that this is not like that, actually, we always send our emails in a normalized email address like `recipient1@example.com` I think depending on the user email provider, we might get an email address like `User Name “User Name <recipient1@example.com>` in notifications.

This caused that we got some backoff events and email bans created for email addresses like `User Name <recipient1@example.com>` , when we send and store a message we normalize the email address before checking if the email address is banned or under a backoff event, so since the format of the email address was different, the email bans and backoff events were ignored when sending a message to these kinds of recipients.

So this caused another issue, since email bans and backoff events were ignored, we continue sending messages to these email addresses that were supposed to be banned or under a backoff event.

In the case of a backoff event, we continue receiving soft bounces for an email address with the format: `User Name <recipient1@example.com>`  and the backoff event was updated increasing its retry counter and next_retry_datetime, the problem arose when it reached the `max_retry_counter`, since the ban was being ignored, it triggered a message retry to the email address that was totally broken, so that started a loop of receiving a `soft_bounce notification` → `enqueue failed message`, since the ban was ignored and the backoff event don’t continue increasing it’s retry_counter and next_retry_datetime because it was previously reached, the result was that we sent approximately 500 emails to this user :cold_sweat:, all of them were bounced.

So to avoid this could happen again in the future I made the following changes:

- Normalize the recipient's email address for every SES notification, bounce, complaint, and delivery, so if we receive an email address like `User Name <recipient1@example.com>`  we convert to `recipient1@example.com` 
- When we reach the max_retry_counter in addition to banning the email address, we delete the backoff event, so in case we ever delete a ban for an email address (maybe a special user request), we’ll avoid a similar issue could happen again.
- Changed manually from admin the current backoff events that were already created with the format `User Name <recipient1@example.com>` to `recipient1@example.com`

### CODE REVIEW

File: cl/users/email_handlers.py [hunk 1]
```diff
@@ -143,6 +145,10 @@ def handle_soft_bounce(
                                 "notification_subtype": EMAIL_NOTIFICATIONS.MAX_RETRY_REACHED,
                             },
                         )
+                        # After ban email address, delete Backoff Event
```
Comment (mlissner) [inline cl/users/email_handlers.py:148]: Is it worth providing more detail here about why? Maybe reference this PR or explain it briefly in the code?

**Review by mlissner — CHANGES_REQUESTED** [review]
Looks good. 

How did you catch this? 

One small suggestion. 

Thanks!

**Comment by albertisfu** [comment]
Thanks! I've added more details to that comment.

I caught this because I was monitoring the FailedEmail objects in admin since I was doing some tests with this account `testing[arroba]a-islas.com` I bought that domain and linked it to an email inbox where I can control the inbox storage to force soft bounces, so I could make some tests to check if retries worked properly (I was afraid that we might be not receiving delivery notifications), after these test I could confirm that we are enqueueing retries properly and receiving delivery notifications that deactivate backoff events (I might write a report about these tests as well), so when checking these tests I saw many FailedEmail objects for that email address (you can see them right now on page 5 of FailedEmail admin), so I analyzed the data about that account and could reproduce the issue locally to fix it.

File: cl/users/email_handlers.py (hunk 1 above)
Comment (albertisfu) [reply cl/users/email_handlers.py]: Sure, added more details and reference to this PR as well.

File: cl/users/email_handlers.py [hunk 2]
```diff
@@ -143,6 +145,15 @@ def handle_soft_bounce(
                                 "notification_subtype": EMAIL_NOTIFICATIONS.MAX_RETRY_REACHED,
                             },
                         )
+                        # After ban an email address, delete Backoff Event in
+                        # case we ever delete a ban for an email address if the
+                        # email address fails again we'll create a new backoff
+                        # event and start the retry counter again avoiding
+                        # unexpected behavior like the one described here:
+                        # https://github.com/freelawproject/courtlistener/pull/2115
```
Comment (mlissner) [inline cl/users/email_handlers.py:153]:
```suggestion
                        # After ban an email address, delete Backoff Event. 
                        # This way, if we delete the ban on the email address,
                        # the backoff event gets a fresh start.
                        # https://github.com/freelawproject/courtlistener/pull/2115
```

**Pull request merged**